sources/updater/updog/src/main.rs

```diff
@@ -437,6 +439,23 @@ fn output<T: Serialize>(json: bool, object: T, string: &str) -> Result<()> {
     Ok(())
 }
 
+fn initiate_reboot() -> Result<()> {
+    // Set up signal handler for termination signals
+    let signals = Signals::new(&[SIGINT, SIGTERM, SIGQUIT]).context(error::Signal)?;
+    thread::spawn(move || {
+        for _sig in signals.forever() {
+            // Ignore termination signals in case updog gets terminated
+            // before getting to exit normally by itself after invoking
+            // `shutdown -r` to complete the update.
+        }
+    });
+    process::Command::new("shutdown")
+        .arg("-r")
+        .status()
+        .context(error::RebootFailure)?;
```

Comment:
If we fail to call the shutdown binary, technically we should kill the signal thread.  I think we can get away without it because we know (currently) the code just prints a backtrace in that case, but it'd be nice to prevent future confusion.